bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/retriever/AbstractCategoryRetriever.java

```diff
@@ -81,6 +81,28 @@
      */
     protected int pageSize = 6;
 
+    /**
+     * Generates the filter for a category or categoryList query
+     *
+     * @param identifiersFilter The filter portion to be used by CategoryFilterInput.
+     * @param categoryIdentifierType The category identifiers used byt the identifiersFilter
+     */
+    public static CategoryFilterInput generateCategoryFilter(FilterEqualTypeInput identifiersFilter,
+        UrlProvider.CategoryIdentifierType categoryIdentifierType) {
+        CategoryFilterInput filter;
+
+        if (UrlProvider.CategoryIdentifierType.UID.equals(categoryIdentifierType)) {
```

Comment:
Am I right that you are missing the `URL_PATH` here. The previous, changed code was a switch

```
switch (categoryIdentifierType) {
            case ID:
                filter = new CategoryFilterInput().setIds(identifierFilter);
                break;
            case UID:
                filter = new CategoryFilterInput().setCategoryUid(identifierFilter);
                break;
            case URL_PATH:
                filter = new CategoryFilterInput().setUrlPath(identifierFilter);
                break;
            default:
                throw new RuntimeException("Category identifier type is not supported");
        }
```